internal/command/arguments/apply.go

```diff
@@ -43,6 +43,9 @@ func ParseApply(args []string) (*Apply, tfdiags.Diagnostics) {
 	}
 
 	cmdFlags := extendedFlagSet("apply", apply.State, apply.Operation, apply.Vars)
+	if apply.State != nil {
+		diags = append(diags, tfdiags.SimpleWarning("state is deprecated"))
```

Comment:
It'd be nice if this error message could present an alternative instead.

The correct thing to do is update the configuration to point to the chosen state file: https://github.com/hashicorp/terraform/issues/35562#issuecomment-2289437736

Maybe we could make this a proper diagnostic with the detail stating to use the path variable in the local backend?

Reply:
Sure, will update and add a commit. And also, I have found a better way to check if the `state` flag is specified, will add this also in the next commit.

Reply:
Hi @liamcervante 
![image](https://github.com/user-attachments/assets/d576ccc7-3945-472c-a9d0-a0ba5172e4d6)
Does it align with your thoughts? I also added the link to the resource for easier navigation.